spring-boot/src/test/kotlin/org/springframework/boot/SpringApplicationExtensionsTests.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.boot
+
+import org.junit.Assert.*
+import org.junit.Test
+import org.springframework.beans.factory.getBean
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+/**
+ * Tests for `SpringApplicationExtensions`.
+ *
+ * @author Sebastien Deleuze
+ */
+class SpringApplicationExtensionsTests {
+
+	@Test
+	fun `Kotlin run() top level function`() {
+		val context = run<ExampleWebConfig>()
+		assertNotNull(context)
+	}
+
+	@Test
+	fun `Kotlin run(arg1, arg2) top level function`() {
+		val context = run<ExampleWebConfig>("--debug", "spring", "boot")
+		val args = context.getBean<ApplicationArguments>()
+		assertArrayEquals(arrayOf("spring", "boot"), args.nonOptionArgs.toTypedArray())
+		assertTrue(args.containsOption("debug"))
+	}
+
+	@Test
+	fun `Kotlin run(array of KClass) top level function`() {
+		val context = run(arrayOf(ExampleConfig::class, ExampleWebConfig::class))
+		assertNotNull(context)
+	}
+
+	@Test
+	fun `Kotlin run(array of KClass, arg1, arg2) top level function`() {
+		val context = run(arrayOf(ExampleConfig::class, ExampleWebConfig::class), "--debug", "spring", "boot")
+		val args = context.getBean<ApplicationArguments>()
+		assertArrayEquals(arrayOf("spring", "boot"), args.nonOptionArgs.toTypedArray())
+		assertTrue(args.containsOption("debug"))
+	}
+
+
+	@Configuration
+	internal open class ExampleConfig
+
+	@Configuration
+	internal open class ExampleWebConfig {
+
+		@Bean
+		open fun webServer(): TomcatServletWebServerFactory {
+			return TomcatServletWebServerFactory(0)
```

Comment:
Is there a reason why you need to create a web server for this test? That looks a bit overkill to me.

Reply:
Yes, because if not the test fails, I just copied what you do in SpringApplicationTests on Java side.